Mage.Sets/src/mage/cards/c/CometStellarPup.java

```diff
@@ -0,0 +1,189 @@
+            ).setTargetPointer(new FixedTargets(token, game)), source);
+        } else if (result == 3) {
+            // [-1]
+            new RemoveCounterSourceEffect(CounterType.LOYALTY.createInstance(1))
+                    .apply(game, source);
+
+            // return a card with mana value 2 or less from your graveyard to your hand.
+            TargetCard target = new TargetCardInYourGraveyard(filterCard);
+            target.setNotTarget(true);
+            if (!target.canChoose(source.getControllerId(), source, game)) {
+                return true;
+            }
+            player.choose(outcome, target, source, game);
+            Card card = game.getCard(target.getFirstTarget());
+            if (card != null) {
+                player.moveCards(card, Zone.HAND, source, game);
+            }
+        } else if (result <= 5) {
+            // Comet, Stellar Pup deals damage equal to the number of loyalty counters on him to a creature or player
+            TargetCreatureOrPlayer target = new TargetCreatureOrPlayer();
+            target.setNotTarget(true);
+            if (!target.canChoose(source.getControllerId(), source, game)) {
+                return true;
+            }
+            player.choose(Outcome.Damage, target, source, game);
+            new DamageTargetEffect(new CountersSourceCount(CounterType.LOYALTY))
+                    .setTargetPointer(new FixedTarget(target.getFirstTarget()))
+                    .apply(game, source);
+
+            // [−2]
+            new RemoveCounterSourceEffect(CounterType.LOYALTY.createInstance(2))
+                    .apply(game, source);
+        } else {
+            //[+1]
+            new AddCountersSourceEffect(CounterType.LOYALTY.createInstance(1))
+                    .apply(game, source);
+
+            // You may activate Comet, Stellar Pup’s loyalty ability two more times this turn.
+            Permanent comet = game.getPermanent(source.getSourceId());
+            if (comet != null) {
+                game.addEffect(
+                        new CometStellarPupContinousEffect(new MageObjectReference(comet, game)),
+                        source
+                );
+            }
+        }
+        return true;
+    }
+}
+
+class CometStellarPupContinousEffect extends ContinuousEffectImpl {
+
+    private final MageObjectReference cometMOR;
+
+    public CometStellarPupContinousEffect(MageObjectReference cometMOR) {
+        super(Duration.EndOfTurn, Layer.RulesEffects, SubLayer.NA, Outcome.Benefit);
+        this.cometMOR = cometMOR;
+    }
+
+    private CometStellarPupContinousEffect(final CometStellarPupContinousEffect effect) {
+        super(effect);
+        this.cometMOR = effect.cometMOR;
+    }
+
+    @Override
+    public CometStellarPupContinousEffect copy() {
+        return new CometStellarPupContinousEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
```

Comment:
does this comply with the ruling?

> In some unusual cases, Comet may gain loyalty abilities other than the one he normally has. In those cases, the effect of rolling a 6 applies only to his normal ability. (2022-10-07)

Reply:
No, I did not see that ruling. Are [[Oath of Teferi]], [[The Chain Veil]] or [[Teferi, Master of Time]] not having that ruling due to being worded differently, and not being linked abilities?

It is more complex to make it so, involving a Watcher and permission specific to Comet's ability.

Reply:
I don't mind leaving it as is

Reply:
Ok let me rebase and clean up.